src/pages/tBTC/Bridge/DepositDetails.tsx

```diff
@@ -135,14 +135,15 @@ export const DepositDetails: PageComponent = () => {
     if (!confirmations || !requiredConfirmations || shouldStartFromFirstStep)
       return
 
-    setMintingProgressStep(
-      getMintingProgressStep({
-        confirmations,
-        requiredConfirmations,
-        optimisticMintingFinalizedTxHash,
-        optimisticMintingRequestedTxHash,
-      })
-    )
+    const step = getMintingProgressStep({
+      confirmations,
+      requiredConfirmations,
+      optimisticMintingFinalizedTxHash,
+      optimisticMintingRequestedTxHash,
+    })
+
+    setMintingProgressStep(step)
+    updateState("depositStep", step)
```

Comment:
Ideally we would like to clear `depositStep` value from the redux store on the `DepositDetails` component unmount. Maybe we could add a return funciton here that would do that?

```
useEffect(() => {
  (...)
  updateState("depositStep", step)
  return () => {
    updateState("depositStep", undefined)
  }
}, [
  /* dependencies */
])
```

Please test it before implementing 😉